### PR TITLE
add an option in insertGpuAllocs to skip the func args copy

### PR DESCRIPTION
**Pull request by zhczhong**
Add an option in insertGpuAllocs to skip the func args copy so that the user could decide whether a gpu::memcopy is needed for the input args

Please review these guidelines to help with the review process:
- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
- [x] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: lib/Transforms/InsertGPUAllocs.cpp [hunk 1]
```diff
@@ -545,7 +551,7 @@ class InsertGPUAllocsPass final
     // with gpu.alloc and insert a memref.copy from host to device
     for (const auto &it : gpuBufferParams) {
       auto param = block.getArgument(it.first);
-      if (isGpuAddrSpace(param))
+      if (isGpuAddrSpace(param) || isUsmArgs.getValue())
```
Comment (AndreyPavlenko) [inline lib/Transforms/InsertGPUAllocs.cpp:554]: I think would be better to put the entire loop into `if (!isUsmArgs.getValue()) { for (...`.

**Comment by AndreyPavlenko** [comment]
There is also pre-commit check failed.

File: lib/Transforms/InsertGPUAllocs.cpp (hunk 1 above)
Comment (zhczhong) [reply lib/Transforms/InsertGPUAllocs.cpp]: Thanks for the advice! I move the `if` out of the loop.(56cf8f1)

**Comment by zhczhong** [comment]
> There is also pre-commit check failed.

The pre-commit check failure is not related to this PR but to other unverified commits. I fix them by [c7be8fc](https://github.com/intel/mlir-extensions/pull/887/commits/c7be8fc9b40ca5f720b23437cec0888dcd85257b)

**Review by AndreyPavlenko — APPROVED** [review]

**Pull request merged**